### ISSUE TITLE
Upgrade automation test step fixes

### ISSUE THREAD

**Issue by shuklanirdesh82** [vmware/vsphere-storage-for-docker#1697]
Reference for steps: https://github.com/vmware/docker-volume-vsphere/issues/1380#issuecomment-312350229

Pre-upgrade test: https://github.com/vmware/docker-volume-vsphere/blob/master/tests/e2e/pre_upgrade_test.go

Pre-upgrade test is not fully automated as planned and following steps were missed during automation task (during initial checkin)

```
6. attach the volume again and leave it as **attached** > just make sure dockerCLI and adminCLI says it is attached
7. Read the data and validate the written data as part of step#3 is intact
```

while talking with @pdhamdhere about one of the customer issue, came across following correction as well.

1. There is no need for restarting docker after upgrading our plugin
2. Volumes stays attached even after upgrade that means our post upgrade test step needs to be fixed too.

Action item:

- [ ] 1. Automate missing steps as shown below
```
6. attach the volume again and leave it as **attached** > just make sure dockerCLI and adminCLI says it is attached
7. Read the data and validate the written data as part of step#3 is intact
```

- [ ] 2. Remove `docker service restart` step from upgrade script (no need for the following step)
https://github.com/vmware/docker-volume-vsphere/blob/master/misc/scripts/upgrade_test.sh#L110
```
$SSH $VM1 "systemctl restart docker || service docker restart"
```

- [ ] 3. Correct post-upgrade steps #1-#4 (https://github.com/vmware/docker-volume-vsphere/blob/master/tests/e2e/post_upgrade_test.go)
```
*Post-Upgrade* test scenario:

*Following steps validates pre-upgrade setup is intact*
1. verify the pre-upgrade setup is intact that means "preUpgradeVolume" should be present
2. Perform verification
VM1: verify using "docker volume ls" & "docker volume inspect" for *attached* status
E1: using 'vmdkops_admin volume ls' to check for *attached* status
3. Read the data and validate the written data as part of step#3 of *Pre-Upgrade* is intact/same as written during pre-upgrade step
4. add some more data to make sure volume is writable with out any problem > perform validation as mentioned in step#2 for *attached* status now
```
In short existing automated (https://github.com/vmware/docker-volume-vsphere/blob/master/tests/e2e/post_upgrade_test.go) test needs to be corrected for the first 4 steps.

/CC @pdhamdhere @ashahi1

**Comment by shuklanirdesh82** [comment]
correctness issue is the mustfix
